syntax = "proto3";

package iroha.protocol;

import "block.proto";

enum ResponseCode {
  OK = 0;
  FAIL = 1;
}

message ToriiResponse {}

message VerifyResponse {
  ResponseCode code = 1;
}

message QueryResponse {}

message QueueTransactionResponse {
  ResponseCode code = 1;
}

service CommandService {
  rpc Torii (Transaction) returns (ToriiResponse);
}

service QueryService {
//  rpc Find (QueryRequest) returns (QueryResponse);
}

service SumeragiService {
  rpc Verify (Block) returns (QueueTransactionResponse);
}

service OrderingService {
  rpc QueueTransaction (Transaction) returns (QueueTransactionResponse);
}
